pkg/pub/manifest.go

```diff
@@ -0,0 +1,172 @@
+package pub
+
+import (
+	"errors"
+	"strings"
+)
+
+// Manifest Main structure for a publication
+type Manifest struct {
+	Context         []string `json:"@context,omitempty"`
+	Metadata        Metadata `json:"metadata"`
+	Links           []Link   `json:"links"`
+	ReadingOrder    []Link   `json:"readingOrder,omitempty"`
+	Resources       []Link   `json:"resources,omitempty"` //Replaces the manifest but less redundant
+	TableOfContents []Link   `json:"toc,omitempty"`
+
+	Subcollections map[string][]PublicationCollection `json:"-"` //Extension point for collections that shouldn't show up in the manifest
```

Comment:
What do you mean by shouldn't show up in the manifest?

For example the `landmarks` collection role from the EPUB profile is a useful subcollection to get for reading apps: https://github.com/readium/webpub-manifest/blob/master/profiles/epub.md

Reply:
Those are not my comments/exclusion behavior, they were kept from the previous code: https://github.com/readium/go-toolkit/blob/0bed5dd41b94da5eca1f904f6346877136134db3/models/publication.go#L27 . I have not yet touched manifest.go beyond the bare minimum (will be doing for parser).

Reply:
Okay, so for the next PR `Internal` can go as well as all the helper functions. Only these two are useful at this point:

https://github.com/readium/kotlin-toolkit/blob/b33a5dd5973aa4a896927df8878b82d03d42b8ee/readium/shared/src/main/java/org/readium/r2/shared/publication/Manifest.kt#L46-L58

Some of the helper functions will be added to the `Publication` struct.